RoyalRoad.java

```diff
@@ -7,60 +7,94 @@
 import java.util.*;
 import java.text.*;
 
-public class OneMax extends FitnessFunction{
-
+public class RoyalRoad extends FitnessFunction {
 /*******************************************************************************
 *                            INSTANCE VARIABLES                                *
 *******************************************************************************/
-
-
-/*******************************************************************************
-*                            STATIC VARIABLES                                  *
-*******************************************************************************/
-
+    private List<String> patterns;
+    private int reward;
+    private String variant;
+    private int buildingBlockSize = 8;
```

Comment:
Should we make buildingBlockSize a parameter so we can double up on our question 2 by increasing the length of each individual and also shorten the length of each individual gene?